src/main/java/org/reactome/release/downloaddirectory/GenerateGOAnnotationFile/CellularComponentAnnotationBuilder.java

```diff
@@ -2,88 +2,180 @@
+        return InstanceUtilities.followInstanceAttributes(
+            reactionlikeEvent, getClassAttributeInstructionsToFollow(), outClasses
+        );
+    }
+
+    private static List<ClassAttributeFollowingInstruction> getClassAttributeInstructionsToFollow() {
+        List<ClassAttributeFollowingInstruction> classesToFollow = new ArrayList<>();
+        for (String className : getClassToClassAttributesMapForObtainingProteins().keySet()) {
+            List<String> classAttributes = getClassToClassAttributesMapForObtainingProteins().get(className);
+            classesToFollow.add(getClassAttributeFollowingInstruction(className, classAttributes));
+        }
+        return classesToFollow;
+    }
+
+    private static Map<String, List<String>> getClassToClassAttributesMapForObtainingProteins() {
+        Map<String, List<String>> classToClassAttributesForFollowingInstructions = new HashMap<>();
+
+        classToClassAttributesForFollowingInstructions.put(
+            ReactomeJavaConstants.Pathway,
+            Arrays.asList(ReactomeJavaConstants.hasEvent)
+        );
+        classToClassAttributesForFollowingInstructions.put(
+            ReactomeJavaConstants.ReactionlikeEvent,
+            Arrays.asList(
+                ReactomeJavaConstants.input, ReactomeJavaConstants.output, ReactomeJavaConstants.catalystActivity
+            )
+        );
+        classToClassAttributesForFollowingInstructions.put(
+            ReactomeJavaConstants.Reaction,
+            Arrays.asList(
+                ReactomeJavaConstants.input, ReactomeJavaConstants.output, ReactomeJavaConstants.catalystActivity
+            )
+        );
+        classToClassAttributesForFollowingInstructions.put(
+            ReactomeJavaConstants.CatalystActivity,
+            Arrays.asList(ReactomeJavaConstants.physicalEntity)
+        );
+        classToClassAttributesForFollowingInstructions.put(
+            ReactomeJavaConstants.Complex,
+            Arrays.asList(ReactomeJavaConstants.hasComponent)
+        );
+        classToClassAttributesForFollowingInstructions.put(
+            ReactomeJavaConstants.EntitySet,
+            Arrays.asList(ReactomeJavaConstants.hasMember)
+        );
+        classToClassAttributesForFollowingInstructions.put(
+            ReactomeJavaConstants.Polymer,
+            Arrays.asList(ReactomeJavaConstants.repeatedUnit)
+        );
+
+        return classToClassAttributesForFollowingInstructions;
+    }
+
+    private static ClassAttributeFollowingInstruction getClassAttributeFollowingInstruction(
+        String className, List<String> classAttributes
+    ) {
+        final List<String> reverseClassAttributes = Collections.emptyList();
+
+        return new ClassAttributeFollowingInstruction(className, classAttributes, reverseClassAttributes);
+    }
+
+    private static String getAnyIssueForCellularComponentDisqualification(GKInstance protein) throws Exception {
```

Comment:
Maybe some documentation explaining what would disqualify a cellular component. It's a little hard to trace, some comments would be nice.

Reply:
Added comments explaining that disqualification for "Cellular Component" has an additional condition to normal disqualification.